arbnode/node.go

```diff
@@ -1578,3 +1584,65 @@ func (n *Node) ExpectChosenSequencer() containers.PromiseInterface[struct{}] {
 func (n *Node) BlockMetadataAtMessageIndex(msgIdx arbutil.MessageIndex) containers.PromiseInterface[common.BlockMetadata] {
 	return containers.NewReadyPromise(n.TxStreamer.BlockMetadataAtMessageIndex(msgIdx))
 }
+
+func (n *Node) GetL1Confirmations(ctx context.Context, msgIdx arbutil.MessageIndex) containers.PromiseInterface[uint64] {
```

Comment:
A function shouldn't both accept context and return a Promise. Caller should use a context to await on a promise, not pass context to this function.
See containers.DoPromise, or, possibly better here, stopWaitet's LaunchPromiseThread (using the L1Reader if parent chain is arbitrum)
In case of IsParentChainArbitrum - parent context should be taken from the

Reply:
Added the LaunchPromiseThread approach